apps/api-extractor/src/enhancers/ValidationEnhancer.ts

```diff
@@ -30,32 +31,56 @@ export class ValidationEnhancer {
     }
   }
 
-  private static _checkForInternalUnderscore(collector: Collector, collectorEntity: CollectorEntity,
-    astSymbol: AstSymbol): void {
-
-    const astSymbolMetadata: SymbolMetadata = collector.fetchMetadata(astSymbol);
-
-    if (astSymbolMetadata.releaseTag === ReleaseTag.Internal && !astSymbolMetadata.releaseTagSameAsParent) {
-      for (const exportName of collectorEntity.exportNames) {
-        if (exportName[0] !== '_') {
-          collector.messageRouter.addAnalyzerIssue(
-            ExtractorMessageId.InternalMissingUnderscore,
-            `The name "${exportName}" should be prefixed with an underscore`
-            + ` because the declaration is marked as @internal`,
-            astSymbol,
-            { exportName }
-          );
+  private static _checkForInternalUnderscore(
+    collector: Collector,
+    collectorEntity: CollectorEntity,
+    astSymbol: AstSymbol
+  ): void {
+    let containsInternal: boolean = false;
+    for (let i: number = 0; i < astSymbol.astDeclarations.length; i++) {
```

Comment:
(optional) It's more idiomatic to write `for (const astDeclaration of astDeclarations) {`

Reply:
I'm going to keep this one. I use the "i" value in the first if-statement block to check if an internal flag was found after non-internal flags

Reply:
We could get really fancy and use destructoring with entries() if our build settings support it. 😄https://flaviocopes.com/how-to-get-index-in-for-of-loop/